data.js

```diff
@@ -56,3 +56,122 @@ const usersArray = [
     gender: 'female'
   }
 ];
+
+
+// Iteration 1: for...of loop
+const getFirstNames = arr => {
+  const userFirstNames = [];
+  for(let user of arr){
+    let firstName = user.firstName;
+    userFirstNames.push(firstName);
+  }
+
+  console.log(userFirstNames)
+  return userFirstNames; 
+};
+
+getFirstNames(usersArray);
+
+
+
+// Iteration 2: for...of loop and ES6 string literals ${}
+const getFullNames = arr => {
+  const userFullName = [];
+  for(let user of arr){
+    let fullName = `${user.firstName} ${user.lastName}`;
+    userFullName.push(fullName);
+  }
+
+  console.log(userFullName);
+  return userFullName;
+};
+
+getFullNames(usersArray)
+
+
+
+// Iteration 3: ES6 destructuring, for of loop, object literal
+const getUsersCreditDetails = arr => {
+  const usersCreditDetails = [];
+  for(let user of arr) {
+    const {firstName,lastName,balance} = user;
+    let userDetails = {firstName, lastName, balance};
+    usersCreditDetails.push(userDetails);
+  }
+
+  console.log(usersCreditDetails);
+  return usersCreditDetails;
+};
+
+getUsersCreditDetails(usersArray);
+
+
+
+
+// Iteration 4: practice .filter() method and how to return two elements
+const genderView = users => {
+  const femaleUsers = []; 
+  const maleUsers = []; 
+  
+  users.filter(user => {
+    if(user.gender === 'female') {
+      femaleUsers.push(`${user.firstName} ${user.lastName}`);
+    } else {
+      maleUsers.push(`${user.firstName} ${user.lastName}`);
+    }
+  });
+
```

Comment:
this line does’t really do what is intending to.
users.filter is not returning anything, but its pushing stuff, so its functioning more like a for each.
ideally you could have done this:
```js
let femaleUsers = users.filter(item => item.gender === “female”)
let maleUsers = users.filter(item => item.gender === “male”)
```
and here you would have the arrays already split by gender. and then you could do the transformation you want by mapping over them or something else.

The filter method you were doing, even though it probably did what you were expecting, it was not the goal of a filter.

The filter must always return a value -> it must be assigned to a variable. Here its not only returning, (in the body of the function) and not being assigned anywhere